sale_procurement_group_by_requested_date/models/sale_order.py

```diff
@@ -9,12 +9,13 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    def _prepare_order_line_procurement(self, cr, uid, order, line,
-                                        group_id=False, context=None):
-        values = super(SaleOrder, self)._prepare_order_line_procurement(
-            cr, uid, order, line, group_id=group_id, context=context)
-        if line.requested_date:
-            req_datetime = fields.Datetime.from_string(line.requested_date)
+    @api.multi
+    def _prepare_order_line_procurement(self, group_id=False):
+        values = super(SaleOrder, self).\
+            _prepare_order_line_procurement(group_id=group_id)
+        self.ensure_one()
+        if self.requested_date:
```

Comment:
previously it was `line.requested_date:` I think that was refering to `sale.order.line` while `self` here is `sale.order`. Also, are you sure about removing `order` and `line` from the imputs? they might be needed in the super call.